core-bundle/src/Security/TwoFactor/FrontendAuthenticationSuccessHandler.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\Security\TwoFactor;
+
+use Contao\User;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\Security\Core\Authentication\Token\TokenInterface;
+use Symfony\Component\Security\Core\Security;
+use Symfony\Component\Security\Http\Authentication\AuthenticationSuccessHandlerInterface;
+use Symfony\Component\Security\Http\HttpUtils;
+
+class FrontendAuthenticationSuccessHandler implements AuthenticationSuccessHandlerInterface
+{
+    /**
+     * @var HttpUtils
+     */
+    private $httpUtils;
+
+    public function __construct(HttpUtils $httpUtils)
+    {
+        $this->httpUtils = $httpUtils;
+    }
+
+    public function onAuthenticationSuccess(Request $request, TokenInterface $token)
+    {
+        /** @var User $user */
+        $user = $token->getUser();
+        $request->getSession()->remove(Security::AUTHENTICATION_ERROR);
+
+        // Enable 2FA
+        $user->useTwoFactor = '1';
```

Comment:
Should we make sure this is a `Contao\FrontendUser`?

Reply:
Definitely.

Reply:
I don't think this needs to be checked, the handler is registered on the frontend firewall.